rpc/blockchain.go

```diff
@@ -173,8 +173,14 @@ func (s *PublicBlockchainService) GetBlockByNumber(
 	blockArgs.InclTx = true
 
 	blockNum := blockNumber.EthBlockNumber()
-	if blockNum != rpc.LatestBlockNumber && blockNum != rpc.PendingBlockNumber {
-		cacheKey := combineCacheKey(uint64(blockNum), s.version, blockArgs)
+	if blockNum != rpc.PendingBlockNumber {
+		var realBlockNum uint64 = 0
```

Comment:
A minor change. Just initialize the `realBlockNum` here and avoid another assignment and branch in line 180-182.
```golang
realBlockNum := uint64(blockNum)
```

Reply:
@AlexiaChen please fix so we can merge

Reply:
Fixed and add some debug log to test in The PR